### ISSUE TITLE
Non-modal DIALOG is broken

### ISSUE THREAD

**Issue by phoe** [Shinmera/qtools-ui#18]
In `dialog.lisp`:
```common-lisp
(defgeneric show (dialog &key modal &allow-other-keys)
  (:method ((dialog dialog) &key (modal T))
    (= (if modal
           (q+:exec dialog)
           (q+:open dialog))
       (q+:qdialog.accepted))))
```

[QDialog::open()](https://doc.qt.io/archives/qt-4.8/qdialog.html#open) returns `void` and therefore its result is not comparable via `#'=`.

**Comment by Shinmera** [comment]
Yeah, this can't work as intended in any case as a non-modal dialog in Qt executes asynchronously. Even if we polled for the dialog to close this could potentially end up blocking the event loop and thus block the application.

I see the following solutions:
1. Add a closure that acts as a callback when the dialog completes
2. Add a future-like thing that gives a unified interface to check for dialog completion (and potentially other async things)
3. Simply document that it will return immediately with no useful value on non-modal show

**Comment by phoe** [comment]
We don't have to block the event loop if we do our own polling - we can call something stupid in there, like `(loop until (closed-p dialog) do (q+:process-events) (sleep 0.01))`. `(q+:process-events)` should ensure that the event loop keeps on running.

**Comment by Shinmera** [comment]
Can we detect whether we're currently executing from within the event loop?

**Comment by phoe** [comment]
Do we have to? Calling `(q+:process-events qt:*qapplication*)` straight from the REPL returns NIL immediately. Same for `(bt:make-thread (lambda () (q+:process-events qt:*qapplication*)))`.

**Comment by Shinmera** [comment]
Yes, because: "Processes all pending events for the calling thread according to the specified flags until there are no more events to process."

Doing so in a lisp thread seems like UB to me.

**Comment by phoe** [comment]
"Calling this function processes events only for the calling thread."

I'd assume that, from the Qt point of view, a Lisp thread is just another external pthread or something similar. And Qt seems to handle those: 

* https://stackoverflow.com/questions/27947293/using-qt-signals-slots-with-non-qt-threads
* https://stackoverflow.com/questions/16207373/qthreadgetcurrentthread-from-non-qt-thread/16210064#16210064
